backend-shared/util/proxyclient.go

```diff
@@ -222,8 +222,13 @@ func (pc *ProxyClient) DeleteAllOf(ctx context.Context, obj client.Object, opts
 
 // StatusWriter knows how to update status subresource of a Kubernetes object.
 func (pc *ProxyClient) Status() client.StatusWriter {
-	res := pc.InnerClient.Status()
-	return res
+
+	wrapper := ProxyClientStatusWrapper{
```

Comment:
Could you elaborate on why a wrapper is needed?  I'm not quite sure what the issue is from the PR description.

Reply:
For unit tests, this is used to keep track of how many/which calls are made to the fake K8s client. In the case of this test, we use to verify that the status field is not updated.

(But we could likely instead use a mocking framework for this same test, rather than using proxy client, if we wanted.)